go/proof.go

```diff
@@ -163,6 +168,95 @@ func (p *ExistenceProof) calculate(spec *ProofSpec) (CommitmentRoot, error) {
 	return res, nil
 }
 
+func (p *ExclusionProof) Calculate() (CommitmentRoot, error) {
+	if p.GetLeaf() == nil {
+		return nil, errors.New("exclusion proof needs defined LeafOp")
+	}
+	if p.Leaf.PrehashKey != HashOp_NO_HASH {
+		return nil, errors.New("exclusion proof must have leaf with PrehashKey == NO_HASH")
+	}
+	if p.Leaf.PrehashValue != HashOp_NO_HASH {
+		return nil, errors.New("exclusion proof must have leaf with PrehashValue == NO_HASH")
+	}
+
+	// leaf step takes the key and value as input
+	res, err := p.Leaf.Apply(p.ActualPath, p.ActualValueHash)
+	if err != nil {
+		return nil, fmt.Errorf("leaf, %w", err)
+	}
+	// check if the actual value was a placeholder and replace with empty hash
+	// assumes the placeholder value is [32]byte
+	if bytes.Equal(p.ActualValueHash, make([]byte, 32)) {
```

Comment:
Readability NIT: Move `make([]byte, 32)` into a constant in this package or a local var and rename to `smtPlaceholder`